contracts/products/SoteriaCoverageProduct.sol

```diff
@@ -445,59 +393,162 @@ contract SoteriaCoverageProduct is ISoteriaCoverageProduct, ERC721, EIP712, Reen
+        uint256 count = holders_.length;
+        require(count == premiums_.length, "length mismatch");
+        require(count <= policyCount(), "policy count exceeded");
+        uint256 amountToPayTreasury = 0;
+
+        for (uint256 i = 0; i < count; i++) {
+            // skip computation if policy inactive
+            if ( !policyStatus(_policyOf[holders_[i]]) ) continue;
+            require(premiums_[i] <= ( _maxRateNum * _chargeCycle * coverLimitOf(policyOf(holders_[i])) ) / _maxRateDenom, "Charging more than promised maximum rate");
+
+            // If policy holder can pay for premium charged in full
+            if (_accountBalanceOf[holders_[i]] + _rewardPointsOf[holders_[i]] >= premiums_[i]) {
+                
+                // If reward points can cover premium charged in full
+                if (_rewardPointsOf[holders_[i]] >= premiums_[i]) {
+                    _rewardPointsOf[holders_[i]] -= premiums_[i];
+                } else {
+                    uint256 amountDeductedFromSoteriaAccount = premiums_[i] - _rewardPointsOf[holders_[i]];
+                    amountToPayTreasury += amountDeductedFromSoteriaAccount;
+                    _accountBalanceOf[holders_[i]] -= amountDeductedFromSoteriaAccount;
+                    _rewardPointsOf[holders_[i]] = 0;
+                }
+                
+                emit PremiumCharged(holders_[i], premiums_[i]);
+            } else {
+                uint256 partialPremium = _accountBalanceOf[holders_[i]] + _rewardPointsOf[holders_[i]];
+                amountToPayTreasury += _accountBalanceOf[holders_[i]];
+                _rewardPointsOf[holders_[i]] = 0;
+                _deactivatePolicy(holders_[i]); // Difference between manually calling deactivatePolicy() and having _deactivatePolicy() called here, is that the remaining account balance goes to Treasury here instead of being returned to the user
+                emit PremiumPartiallyCharged(holders_[i], premiums_[i], partialPremium);
+            }  
+        }
+        // transfer premium to the treasury
+        ITreasury(payable(_registry.treasury())).routePremiums{value: amountToPayTreasury}();
     }
 
     /***************************************
     INTERNAL FUNCTIONS
     ***************************************/
 
+    /**
+    * @notice Given a request for new coverage, determines if there is sufficient coverage capacity.
+    * @param existingTotalCover_ The existing total cover, will be 0 when a policy is first purchased.
+    * @param newTotalCover_  The total cover amount requested in a new cover request.
+    * @return acceptable True there is sufficient capacity for requested new coverage amount, false otherwise.
+    */
+    function _canPurchaseNewCover(uint256 existingTotalCover_, uint256 newTotalCover_) internal view returns (bool acceptable) {
+        // Redundant check if _activeCoverLimit will be lowered
+        if (newTotalCover_ < existingTotalCover_) return true;
```

Comment:
we remove  **if (newTotalCover_ < existingTotalCover_) return true;** statement.